frame/composable-traits/src/dex.rs

```diff
@@ -118,6 +121,16 @@ pub trait Amm {
 		min_receive: Self::Balance,
 		keep_alive: bool,
 	) -> Result<Self::Balance, DispatchError>;
+
+	fn remove_liquidity_dryrun(
+		who: &Self::AccountId,
+		pool_id: Self::PoolId,
+		lp_amount: Self::Balance,
+		min_base_amount: Self::Balance,
```

Comment:
We need to introduce asset Ids (lists of assets) for these types of calls, names like `base` and `quote` doesn't cut it anymore as we want to move towards multi asset pools.

Reply:
Probably best to have a common data structure that captures an `asset_id` and a `balance`

Reply:
Is if fine to do that in separate change? AFAIK these RPCs are priority for FE.

Reply:
Also I request to have small PRs which has changes related to the subject of given click-up ticket. This allows reviewers to have fast reviews and make overall progress faster. IMHO diverging to other large subject is not good.

Reply:
Unfortunately, there is no design document, but I also explained to you the same requirement before. We have ongoing discussions on supporting multi-asset stable swap soon. So this is not hypothetical. Since assetIds are expected to be provided through these functions, so for now, we can assert the order of assets provided as base first for now. Later we can switch the implementation easily.

Reply:
>Also I request to have small PRs which has changes related to the subject of given click-up ticket.

I understand, but this function signature is being changed in the current PR, so I see that it's not going to cause a change in PR scope.

Reply:
@vimukthi-git 
```
We need to introduce asset Ids (lists of assets) for these types of calls, names like base and quote
``` 
this is future requirements, came after current requirement to get things done. for me it feels by process should be separate PR

Reply:
@dzmitry-lahoda I do not consider this a new requirement nor was it not known before about this concept. You can refer this comment in a previous PR https://github.com/ComposableFi/composable/pull/1093#discussion_r886559668 where this same thing was explained.

Reply:
There is already a task for assessment https://app.clickup.com/t/2gh7cjb . I changed functions to use Vec<> but as we see there is also possibilities of using HashMap. IMHO these kind of discussion should be part of design. Also when we complete 3 pool support in pallet then FE also need to use it. So even for them it is a rework.

Reply:
@KaiserKarel solution makes sense. Only requirement from my end is to prevent us from assuming two assets at the interface level from now until we have a design/tasks to implement multi asset.